osu.Game.Rulesets.Taiko/Mods/TaikoModSimplifiedRhythm.cs

```diff
@@ -0,0 +1,135 @@
+                { 6, 4 }, // 1/6 snap to 1/4 snap
+                { 3, 2 }, // 1/3 snap to 1/2 snap
+            };
+
+            int patternStartIndex = 0;
+            bool inPattern = false;
+
+            List<Hit> hits = taikoBeatmap.HitObjects.Where(obj => obj is Hit).Cast<Hit>().ToList();
+
+            foreach (var snapConversion in snapConversions)
+            {
+                // Skip processing if the corresponding conversion is disabled
+                if (!shouldProcessRhythm(snapConversion.Key))
+                    continue;
+
+                for (int i = 0; i < hits.Count; i++)
+                {
+                    double snapValue = i < hits.Count - 1
+                        ? getSnapBetweenNotes(controlPointInfo, hits[i], hits[i + 1])
+                        : 1; // No next note, default to a safe 1/1 snap
+
+                    if (snapValue == snapConversion.Key)
+                    {
+                        if (!inPattern)
+                        {
+                            patternStartIndex = i;
+                        }
+
+                        inPattern = true;
+                    }
+
+                    // check if end of pattern or if we're on the last note
+                    if ((inPattern && snapValue != snapConversion.Key) || i == hits.Count)
+                    {
+                        // End of the pattern
+                        inPattern = false;
+
+                        // Iterate through the pattern
+                        for (int j = patternStartIndex; j <= i; j++)
+                        {
+                            int currentHitPosition = j - patternStartIndex;
+
+                            if (snapConversion.Key == 8)
+                            {
+                                // 1/8: Remove the second note
+                                if (currentHitPosition % 2 == 1)
+                                {
+                                    toRemove.Add(hits[j]);
+                                }
+                            }
+                            else
+                            {
+                                // 1/6 and 1/3: Adjust the second note and remove the third
+                                if (currentHitPosition % 3 == 1)
+                                {
+                                    hits[j].StartTime = hits[j - 1].StartTime + (controlPointInfo.TimingPointAt(hits[j].StartTime).BeatLength / Convert.ToDouble(snapConversion.Value));
+                                }
+                                else if (currentHitPosition % 3 == 2)
+                                {
+                                    toRemove.Add(hits[j]);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                // Remove queued notes
+                taikoBeatmap.HitObjects = taikoBeatmap.HitObjects.Except(toRemove).ToList();
```

Comment:
```suggestion
                taikoBeatmap.HitObjects.RemoveAll(obj => toRemove.Contains(obj));

```

`Except` creates a new collection, this just modifies the one in place.